modules/@apostrophecms/ui/ui/apos/components/AposCombo.vue

```diff
@@ -88,26 +104,51 @@ export default {
     value: {
       type: Object,
       required: true
+    },
+    typehead: {
+      type: Boolean,
+      default: false
+    },
+    busy: {
+      type: Boolean,
+      default: false
+    },
+    // When true, indicates that the values and selected items are arrays of objects,
+    // array of primitive values otherwise.
+    objectValues: {
+      type: Boolean,
+      default: false
     }
   },
 
-  emits: [ 'select-items' ],
+  emits: [ 'select-items', 'toggle', 'search' ],
   data () {
-    const showSelectAll = this.field.all !== false &&
-      (!this.field.max || this.field.max > this.choices.length);
 
     return {
       showedList: false,
       boxHeight: 0,
-      showSelectAll,
-      options: this.renderOptions(showSelectAll),
       boxResizeObserver: this.getBoxResizeObserver(),
-      focusedItemIndex: null
+      focusedItemIndex: null,
+      thInput: ''
     };
   },
   computed: {
+    showSelectAll() {
+      return this.field.all !== false &&
+        (!this.field.max || this.field.max > this.choices.length);
+    },
+    options() {
+      return this.renderOptions(this.showSelectAll);
+    },
     selectedItems() {
-      if (this.allItemsSelected()) {
+      if (this.showSelectAll && this.allItemsSelected()) {
```

Comment:
[Optional] Could use early return here to avoid nested if statements:
```javascript
    selectedItems() {
      if (!this.showSelectAll && !this.allItemsSelected()) {
        return this.value.data;
      }

      if (this.objectValues) {
        const { listLabel } = this.getSelectAllLabel();
        return [ {
          label: listLabel,
          value: '__all'
        } ];
      }

      return [ '__all' ];
    }
```